src/env.h

```diff
@@ -463,6 +472,12 @@ class Environment {
 
   inline int64_t get_async_wrap_uid();
 
+  // List of id's that have been destroyed and need the destroy() cb called.
+  inline void add_destroy_id(int64_t id);
+  inline int64_t get_destroy_id();
```

Comment:
Naming this `get_…` makes this sound a bit like a getter without side effects. How about `pop_destroy_id()` or something like that? (And if you like, correspondingly `push_destroy_id`?)